cloudinit/config/cc_kernel_modules.py

```diff
@@ -0,0 +1,395 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+"""Kernel Modules"""
+import copy
+import re
+from array import array
+from logging import Logger
+from textwrap import dedent
+from typing import Dict, List, Union
+
+from cloudinit import log as logging
+from cloudinit import subp, util
+from cloudinit.cloud import Cloud
+from cloudinit.config import Config
+from cloudinit.config.schema import MetaSchema, get_meta_doc
+from cloudinit.settings import PER_INSTANCE
+
+MODULE_DESCRIPTION = dedent(
+    """\
+"""
+)
+
+DISTROS = ["debian", "ubuntu"]
+
+meta: MetaSchema = {
+    "id": "cc_kernel_modules",
+    "name": "Kernel Modules",
+    "title": "Module to load/blacklist/enhance kernel modules",
+    "description": MODULE_DESCRIPTION,
+    "distros": DISTROS,
+    "frequency": PER_INSTANCE,
+    "activate_by_schema_keys": ["kernel_modules"],
+    "examples": [
+        dedent(
+            """\
+    kernel_modules:
+      - name: wireguard
+        load: true
+      - name: v4l2loopback
+        load: true
+        persist:
+          options: "devices=1 video_nr=20 card_label=fakecam exclusive_caps=1"
+      - name: zfs
+        persist:
+          blacklist: true
+    """
+        ),
+    ],
+}
+
+__doc__ = get_meta_doc(meta)
+
+LOG = logging.getLogger(__name__)
+NL = "\n"
+REQUIRED_KERNEL_MODULES_KEYS = frozenset(["name"])
+
+DefaultConfigType = Dict[
+    str, Union[Dict[str, List[str]], Dict[str, Dict[str, Union[str, int]]]]
+]
+
+DEFAULT_CONFIG: DefaultConfigType = {
```

Comment:
What about the following? I do not see a lot of value in this complex type, even mypy cannot properly analyze the code: https://github.com/canonical/cloud-init/actions/runs/3465208769/jobs/5787668073

```suggestion
DEFAULT_CONFIG: dict = {
```

Reply:
Agreed :)